### ISSUE TITLE
Will there be any support for .Net Core?

### ISSUE THREAD

**Issue by szabolcsx** [ppatierno/gnatmq#25]

**Comment by guidefreitas** [comment]
I'm also interested in this.  Any answer?

**Comment by VladimirAkopyan** [comment]
I have it running in a UWP app, so I believe it should work for .Net Core if you use the WinRT version

**Comment by guidefreitas** [comment]
It may be compatible but the nuget package declaration does not specify the core framework. Import the nuget package in a .net core application result in the following error:

Errors in c:\users\guilh\documents\visual studio 2015\Projects\MQTTCore\src\MQTTCore\MQTTCore.xproj
    Package GnatMQ 1.1.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package GnatMQ 1.1.0 supports:
      - net45 (.NETFramework,Version=v4.5)
      - netcf39 (netcf39,Version=v0.0)
      - win81 (Windows,Version=v8.1)
      - wpa81 (WindowsPhoneApp,Version=v8.1)
    One or more packages are incompatible with .NETCoreApp,Version=v1.0.

**Comment by VladimirAkopyan** [comment]
it looks like I was wrong. The WinRT version, which is what i've been using on UWP, is definately not compatible with .net core, because it uses the Apis from Windows namespace.

**Comment by guidefreitas** [comment]
Yeah, I think its necessary to compile with .NET Standard library to get the cross platform compatibility for now on.

**Comment by jot4p** [comment]
Hi, 
I need to set username and password.. it's possible ?

**Comment by markusschaber** [comment]
.NET Standard 2.0 brings a compatibility layer for libraries compiled against .NET Framework 4.5, see:
https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/
http://www.talkingdotnet.com/whats-new-in-net-standard-2/
That might allow to run the code on .NET Core.

**Comment by guidefreitas** [comment]
Worked great here using .NET Core 2.0 on OSX. Thanks.
